### PR TITLE
Add Telemetry Docs

### PR DESCRIPTION
**Pull request by breeswish**
### What is changed, added or deleted? (Required)

Add docs about the telemetry functionality introduced in TiDB v4.0.2.

- [x] TODO: Modify tidb-ansible to provide TiDB and PD telemetry config https://github.com/pingcap/tidb-ansible/pull/1319

The following images demostrate how it looks like for the `<details>` tag when rendering on PingCAP docs:

image

image

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: telemetry.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+### 动态禁用 TiDB 遥测
+
+对于已启动的 TiDB 集群，还可以修改系统全局变量 [`tidb_enable_telemetry`](/tidb-specific-system-variables.md#tidb_enable_telemetry) 动态禁用 TiDB 遥测功能：
+
+```sql
+SET GLOBAL tidb_enable_telemetry = 0;
+```
+
+配置文件的禁用优先级高于全局变量。若通过配置文件禁用了遥测功能，则全局变量的配置将不起作用，遥测功能总是处于关闭状态。
```
Comment (july2993) [inline telemetry.md:139]: 如果配置文件开起（虽然不会怎么陪），也是没法用系统变量关掉的对吧

Comment (breeswish) [reply telemetry.md:139]: 配置文件中开启是默认的行为，这个情况下可以用系统变量关闭

File: telemetry.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+# enable-telemetry: true
+```
+
+将其修改为：
+
+```yaml
+enable-telemetry: false
+```
+
+详情参见[使用 Ansible 部署](/online-deployment-using-ansible.md)。
```
Comment (TomShawn) [inline telemetry.md:123]:
```suggestion
详情参见[使用 TiDB Ansible 部署](/online-deployment-using-ansible.md)。
```

File: telemetry.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+</details>
+
+<details>
+  <summary>通过 TiDB Operator 在 Kubernetes 部署</summary>
```
Comment (TomShawn) [inline telemetry.md:128]:
```suggestion
  <summary>通过 TiDB Operator 在 Kubernetes 上部署</summary>
```

File: telemetry.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+SET GLOBAL tidb_enable_telemetry = 0;
+```
+
+配置文件的禁用优先级高于全局变量。若通过配置文件禁用了遥测功能，则全局变量的配置将不起作用，遥测功能总是处于关闭状态。
+
+### 禁用 TiDB Dashboard 遥测
+
+可以修改 PD 配置中 [`dashboard.disable-telemetry = true`](/pd-configuration-file.md#disable-telemetry) 禁用 TiDB Dashboard 遥测功能。对于已启动的集群，该配置需要重启后才能生效。
+
+以下列出了各个部署工具中进行配置修改的具体步骤。
```
Comment (TomShawn) [inline telemetry.md:150]:
```suggestion
以下列出在各个部署工具中修改遥测配置的具体步骤。
```

File: telemetry.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+```yaml
+server_configs:
+  pd:
+    dashboard.disable-telemetry: true
+```
+
+</details>
+
+<details>
+  <summary>通过 Ansible 部署</summary>
```
Comment (TomShawn) [inline telemetry.md:202]:
```suggestion
  <summary>通过 TiDB Ansible 部署</summary>
```

File: telemetry.md [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+
+将其修改为：
+
+```yaml
+dashboard:
+  ...
+  disable-telemetry: true
+```
+
+详情参见[使用 Ansible 部署](/online-deployment-using-ansible.md)。
```
Comment (TomShawn) [inline telemetry.md:220]:
```suggestion
详情参见[使用 TiDB Ansible 部署](/online-deployment-using-ansible.md)。
```

File: telemetry.md [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+
+</details>
+
+<details>
+  <summary>通过 TiDB Operator 在 Kubernetes 部署</summary>
```
Comment (TomShawn) [inline telemetry.md:225]:
```suggestion
  <summary>通过 TiDB Operator 在 Kubernetes 上部署</summary>
```

File: telemetry.md [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+
+对于 TiUP 遥测，可通过执行以下命令查看遥测状态：
+
+```bash
+tiup telemetry status
+```
+
+## 使用情况信息合规性
+
+为了满足不同国家或地区对于此类信息的合规性要求，使用情况信息会按照操作者的 IP 地址的不同发送到位于不同国家服务器，具体如下：
```
Comment (TomShawn) [inline telemetry.md:288]:
```suggestion
为了满足不同国家或地区对于此类信息的合规性要求，使用情况信息会按照不同的操作者 IP 地址发送到位于不同国家的服务器，具体如下：
```

File: telemetry.md [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+
+- 当 IP 为中国大陆 IP 地址时，使用情况信息将会发送并存储于中国大陆境内的公有云服务器。
```
Comment (TomShawn) [inline telemetry.md:290]:
```suggestion
- 若为中国大陆 IP 地址，使用情况信息将会发送并存储于中国大陆境内的公有云服务器。
```

File: telemetry.md [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+- 当 IP 为中国大陆以外 IP 地址时，使用情况信息将会发送并存储于美国的公有云服务器。
```
Comment (TomShawn) [inline telemetry.md:291]:
```suggestion
- 若为中国大陆以外 IP 地址，使用情况信息将会发送并存储于美国的公有云服务器。
```

File: tidb-configuration-file.md [hunk 11]
```diff
@@ -140,6 +140,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 单位：byte。
 + 目前的合法值范围 `[3072, 3072*4]`。MySQL 和 TiDB v3.0.11 之前版本（不包含 v3.0.11）没有此配置项，不过都对新建索引的长度做了限制。MySQL 对此的长度限制为 `3072`，TiDB 在 v3.0.7 以及之前版本该值为 `3072*4`，在 v3.0.7 之后版本（包含 v3.0.8、v3.0.9 和 v3.0.10）的该值为 `3072`。为了与 MySQL 和 TiDB 之前版本的兼容，添加了此配置项。
 
+### `enable-telemetry` <span class="version-mark">从 v4.0.2 版本开始引入</span>
+
++ 是否开启 TiDB 遥测功能。
++ 默认值：true
++ 如果所有 TiDB 实例上该选项都设置为 `false`，那么将完全禁用 TiDB 遥测功能，且忽略 [`tidb_enable_telemetry`](/tidb-specific-system-variables.md#tidb_enable_telemetry) 系统变量。参阅[遥测](/telemetry.md)章节了解该功能详情。
```
Comment (TomShawn) [inline tidb-configuration-file.md:147]:
```suggestion
+ 如果所有 TiDB 实例上该选项都设置为 `false`，那么将完全禁用 TiDB 遥测功能，且忽略 [`tidb_enable_telemetry`](/tidb-specific-system-variables.md#tidb_enable_telemetry) 系统变量。参阅[遥测](/telemetry.md)了解该功能详情。
```

File: tidb-specific-system-variables.md [hunk 12]
```diff
@@ -949,3 +949,11 @@ TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在
...
 
 这个变量设置了查询 METRIC_SCHEMA 时生成的 Prometheus 语句的 range duration，单位为秒。
+
+### tidb_enable_telemetry <span class="version-mark">从 v4.0.2 版本开始引入</span>
+
+作用域：GLOBAL
+
+默认值: 1
+
+这个变量用于动态地控制 TiDB 遥测功能是否开启。设置为 0 可以关闭 TiDB 遥测功能。当所有 TiDB 都有设置 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry) 为 `false` 时将忽略该系统变量、总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)章节了解该功能详情。
```
Comment (TomShawn) [inline tidb-specific-system-variables.md:959]:
```suggestion
这个变量用于动态地控制 TiDB 遥测功能是否开启。设置为 `0` 可以关闭 TiDB 遥测功能。当所有 TiDB 实例都设置 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry) 为 `false` 时将忽略该系统变量并总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)了解该功能详情。
```

File: tiup/tiup-overview.md [hunk 13]
```diff
@@ -32,6 +32,10 @@ curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh
 tiup --version
 ```
 
+> **注意：**
+>
+> TiUP 默认会收集使用情况信息，并将其分享给 PingCAP 用于改善产品。若要了解收集的信息详情及如何禁用这个行为，请参见[遥测](/telemetry.md)章节。
```
Comment (TomShawn) [inline tiup/tiup-overview.md:37]:
```suggestion
> TiUP 默认会收集使用情况信息，并将信息分享给 PingCAP 用于改善产品。若要了解所收集的信息详情及如何禁用该收集行为，请参见[遥测](/telemetry.md)。
```

File: upgrade-tidb-using-ansible.md [hunk 14]
```diff
@@ -222,3 +222,7 @@ ansible-playbook local_prepare.yml
 ```bash
 ansible-playbook rolling_update_monitor.yml
 ```
+
+> **注意：**
+>
+> TiDB (v4.0.2 版本起) 默认会定期收集使用情况信息，并将其分享给 PingCAP 用于改善产品。若要了解收集的信息详情及如何禁用这个行为，请参见[遥测](/telemetry.md)章节。
```
Comment (TomShawn) [inline upgrade-tidb-using-ansible.md]: ditto

File: upgrade-tidb-using-tiup.md [hunk 15]
```diff
@@ -190,6 +190,10 @@ TiDB Cluster: <cluster-name>
 TiDB Version: v4.0.0
 ```
 
+> **注意：**
+>
+> TiUP 及 TiDB (v4.0.2 版本起) 默认会收集使用情况信息，并将其分享给 PingCAP 用于改善产品。若要了解收集的信息详情及如何禁用这个行为，请参见[遥测](/telemetry.md)章节。
```
Comment (TomShawn) [inline upgrade-tidb-using-tiup.md]: ditto

File: upgrade-tidb-using-tiup-offline.md [hunk 16]
```diff
@@ -25,3 +25,7 @@ export TIUP_MIRRORS=/path/to/mirror
 ## 2. 升级 TiDB 集群
 
 在更新本地镜像后，可参考[使用 TiUP 升级 TiDB](/upgrade-tidb-using-tiup.md#使用-tiup-升级-tidb)升级 TiDB 集群。
+
+> **注意：**
+>
+> TiUP 及 TiDB (v4.0.2 版本起) 默认会收集使用情况信息，并将其分享给 PingCAP 用于改善产品。若要了解收集的信息详情及如何禁用这个行为，请参见[遥测](/telemetry.md)章节。
```
Comment (TomShawn) [inline upgrade-tidb-using-tiup-offline.md]: ditto

File: quick-start-with-tidb.md [hunk 17]
```diff
@@ -307,3 +307,7 @@ TiDB-Wasm 是运行在浏览器中的 TiDB 数据库，打开网页即可使用
     - 在线部署：[使用 TiUP 部署 TiDB 集群](/production-deployment-using-tiup.md)
     - 离线部署：[使用 TiUP 离线部署 TiDB 集群](/production-offline-deployment-using-tiup.md)
     - [使用 TiDB Operator 在云上部署 TiDB](https://docs.pingcap.com/zh/tidb-in-kubernetes/v1.1)
+
+> **注意：**
+>
+> TiDB、TiUP 及 TiDB Dashboard 默认会收集使用情况信息，并将其分享给 PingCAP 用于改善产品。若要了解收集的信息详情及如何禁用这个行为，请参见[遥测](/telemetry.md)章节。
```
Comment (TomShawn) [inline quick-start-with-tidb.md:313]:
```suggestion
> TiDB、TiUP 及 TiDB Dashboard 默认会收集使用情况信息，并将信息分享给 PingCAP 用于改善产品。若要了解所收集的信息详情及如何禁用该收集行为，请参见[遥测](/telemetry.md)。
```

File: production-offline-deployment-using-tiup.md [hunk 18]
```diff
@@ -264,3 +264,7 @@ tiup cluster start tidb-test
 预期日志结尾输出会有 ```Deployed cluster `tidb-test` successfully``` 关键词，表示部署成功。
 
 部署完成后，集群相关操作可参考 [cluster 命令](/tiup/tiup-cluster.md)。
+
+> **注意：**
+>
+> TiDB 和 TiUP 默认会收集使用情况信息，并将其分享给 PingCAP 用于改善产品。若要了解收集的信息详情及如何禁用这个行为，请参见[遥测](/telemetry.md)章节。
```
Comment (TomShawn) [inline production-offline-deployment-using-tiup.md:270]: ditto

File: pd-configuration-file.md [hunk 19]
```diff
@@ -282,3 +282,19 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
+
+## dashboard
+
+PD 中内置的 [TiDB Dashboard](/dashboard/dashboard-intro.md) 相关配置项。
+
+### `public-path-prefix`
+
++ 通过反向代理访问 TiDB Dashboard 时，配置反向代理提供服务的路径前缀。
++ 默认："/dashboard"
++ 若不通过反向代理访问 TiDB Dashboard，请勿配置该配置项，否则可能导致 TiDB Dashboard 无法正常访问。关于该配置的详细使用场景，参见 [通过反向代理使用 TiDB Dashboard](/dashboard/dashboard-ops-reverse-proxy.md)。
```
Comment (TomShawn) [inline pd-configuration-file.md:294]:
```suggestion
+ 若不通过反向代理访问 TiDB Dashboard，**请勿配置该配置项**，否则可能导致 TiDB Dashboard 无法正常访问。关于该配置的详细使用场景，参见[通过反向代理使用 TiDB Dashboard](/dashboard/dashboard-ops-reverse-proxy.md)。
```

File: telemetry.md [hunk 20]
```diff
@@ -0,0 +1,283 @@
+---
+title: 遥测
+category: reference
+---
+
+# 遥测
+
+TiDB、TiUP 及 TiDB Dashboard 默认会收集使用情况信息，并将其分享给 PingCAP 用于改善产品，例如，通过这些使用情况信息，PingCAP 可以了解常见的 TiDB 集群规模，从而确定新功能优先级。
```
[the remark this answers is not in the document]
Comment (winkyao) [reply telemetry.md:8]: 例如，通过这些使用情况信息，PingCAP 可以了解常见的 TiDB 集群操作，从而确定新功能优先级。

File: pd-configuration-file.md [hunk 21]
```diff
@@ -282,3 +282,19 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
...
+
+## dashboard
+
+PD 中内置的 [TiDB Dashboard](/dashboard/dashboard-intro.md) 相关配置项。
+
+### `public-path-prefix`
+
++ 通过反向代理访问 TiDB Dashboard 时，配置反向代理提供服务的路径前缀。
++ 默认："/dashboard"
++ 若不通过反向代理访问 TiDB Dashboard，**请勿配置该配置项**，否则可能导致 TiDB Dashboard 无法正常访问。关于该配置的详细使用场景，参见[通过反向代理使用 TiDB Dashboard](/dashboard/dashboard-ops-reverse-proxy.md)。
```
Comment (TomShawn) [inline pd-configuration-file.md:294]:
```suggestion
+ 若不通过反向代理访问 TiDB Dashboard，**请勿配置该项**，否则可能导致 TiDB Dashboard 无法正常访问。关于该配置的详细使用场景，参见[通过反向代理使用 TiDB Dashboard](/dashboard/dashboard-ops-reverse-proxy.md)。
```

File: telemetry.md [hunk 22]
```diff
@@ -0,0 +1,295 @@
...
+category: reference
+---
+
+# 遥测
+
+TiDB、TiUP 及 TiDB Dashboard 默认会收集使用情况信息，并将这些信息分享给 PingCAP 用于改善产品，例如，通过这些使用情况信息，PingCAP 可以了解常见的 TiDB 集群操作，从而确定新功能优先级。
+
+## 哪些使用情况信息会被收集？
+
+以下具体描述了各个组件收集并分享的使用情况信息。若收集的使用情况信息有变化，将在版本更新说明中告知。
```
Comment (TomShawn) [inline telemetry.md:13]:
```suggestion
以下章节具体描述了各个组件收集并分享的使用情况信息。若收集的使用情况信息有变化，将在版本更新说明中告知。
```

File: telemetry.md [hunk 23]
```diff
@@ -0,0 +1,295 @@
...
+
+当 TiUP 遥测功能开启时，执行 TiUP 命令时将会将使用情况信息分享给 PingCAP，包括（但不限于）：
+
+- 随机生成的遥测标示符
+- TiUP 命令的执行情况，如命令执行是否成功、命令执行耗时等
+- 使用 TiUP 进行部署的情况，如部署的目标机器硬件信息、组件版本号、修改过的部署配置名称等
+
+使用 TiUP 时，可通过设置 `TIUP_CLUSTER_DEBUG=enable` 环境变量输出执行命令时收集的使用情况信息，例如：
+
+{{< copyable "shell" >}}
```
Comment (TomShawn) [inline telemetry.md:54]:
```suggestion
{{< copyable "shell-regular" >}}
```

File: telemetry.md [hunk 24]
```diff
@@ -0,0 +1,295 @@
...
+部署 TiDB 集群时，可以为每个 TiDB 集群设置 [`enable-telemetry = false`](/tidb-configuration-file.md#enable-telemetry) 以禁用 TiDB 遥测功能。也可以在已部署的 TiDB 集群上修改该配置项，但需要重启集群后才能生效。
+
+以下是在各个部署工具中修改遥测配置的具体步骤。
+
+<details>
+  <summary>通过二进制手工部署</summary>
+
+创建配置文件 `tidb_config.toml` 包含如下内容：
+
+{{< copyable "toml" >}}
```
Comment (TomShawn) [inline telemetry.md:73]:
```suggestion
{{< copyable "" >}}
```

File: telemetry.md [hunk 25]
```diff
@@ -0,0 +1,295 @@
...
+详情参见 [TiDB 配置参数](/command-line-flags-for-tidb-configuration.md#--config)、[TiDB 配置文件描述](/tidb-configuration-file.md#enable-telemetry)。
+
+</details>
+
+<details>
+  <summary>通过 TiUP Playground 试用</summary>
+
+创建配置文件 `tidb_config.toml` 包含如下内容：
+
+{{< copyable "toml" >}}
```
Comment (TomShawn) [inline telemetry.md:90]:
```suggestion
{{< copyable "" >}}
```

File: telemetry.md [hunk 26]
```diff
@@ -0,0 +1,295 @@
...
+
+```toml
+enable-telemetry = false
+```
+
+启动 TiUP Playground 时，指定命令行参数 `--db.config tidb_config.toml` 使得该配置生效，如：
+
+{{< copyable "shell" >}}
```
Comment (TomShawn) [inline telemetry.md:98]:
```suggestion
{{< copyable "shell-regular" >}}
```

File: telemetry.md [hunk 27]
```diff
@@ -0,0 +1,295 @@
...
+详情参见 [TiUP - 本地快速部署 TiDB 集群](/tiup/tiup-playground.md)。
+
+</details>
+
+<details>
+  <summary>通过 TiUP Cluster 部署</summary>
+
+修改部署拓扑文件 `topology.yaml`，新增（或在现有项中添加）以下内容：
+
+{{< copyable "yaml" >}}
```
Comment (TomShawn) [inline telemetry.md:113]:
```suggestion
{{< copyable "" >}}
```

File: telemetry.md [hunk 28]
```diff
@@ -0,0 +1,295 @@
...
+可以修改 PD 配置中 [`dashboard.disable-telemetry = true`](/pd-configuration-file.md#disable-telemetry) 禁用 TiDB Dashboard 遥测功能。对于已启动的集群，该配置需要重启后才能生效。
+
+以下列出在各个部署工具中修改遥测配置的具体步骤。
+
+<details>
+  <summary>通过二进制手工部署</summary>
+
+创建配置文件 `pd_config.toml` 包含如下内容：
+
+{{< copyable "toml" >}}
```
Comment (TomShawn) [inline telemetry.md:174]:
```suggestion
{{< copyable "" >}}
```

File: telemetry.md [hunk 29]
```diff
@@ -0,0 +1,295 @@
...
+详情参见 [PD 配置参数](/command-line-flags-for-pd-configuration.md#--config)、[PD 配置文件描述](/pd-configuration-file.md#disable-telemetry)。
+
+</details>
+
+<details>
+  <summary>通过 TiUP Playground 试用</summary>
+
+创建配置文件 `pd_config.toml` 包含如下内容：
+
+{{< copyable "toml" >}}
```
Comment (TomShawn) [inline telemetry.md:192]:
```suggestion
{{< copyable "" >}}
```

File: telemetry.md [hunk 30]
```diff
@@ -0,0 +1,295 @@
...
+
+```toml
+[dashboard]
+disable-telemetry = true
+```
+
+启动 TiUP Playground 时，指定命令行参数 `--pd.config pd_config.toml` 使得该配置生效，如：
+
+{{< copyable "shell" >}}
```
Comment (TomShawn) [inline telemetry.md:201]:
```suggestion
{{< copyable "shell-regular" >}}
```

File: telemetry.md [hunk 31]
```diff
@@ -0,0 +1,295 @@
...
+详情参见 [TiUP - 本地快速部署 TiDB 集群](/tiup/tiup-playground.md)。
+
+</details>
+
+<details>
+  <summary>通过 TiUP Cluster 部署</summary>
+
+修改部署拓扑文件 `topology.yaml`，新增（或在现有项中添加）以下内容：
+
+{{< copyable "yaml" >}}
```
Comment (TomShawn) [inline telemetry.md:216]:
```suggestion
{{< copyable "" >}}
```

File: telemetry.md [hunk 32]
```diff
@@ -0,0 +1,295 @@
...
+
+详情参见[在标准 Kubernetes 上部署 TiDB 集群](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/deploy-on-general-kubernetes)。
+
+</details>
+
+### 禁用 TiUP 遥测
+
+可通过执行以下命令禁用 TiUP 遥测功能：
+
+{{< copyable "shell" >}}
```
Comment (TomShawn) [inline telemetry.md:262]:
```suggestion
{{< copyable "shell-regular" >}}
```

File: telemetry.md [hunk 33]
```diff
@@ -0,0 +1,295 @@
...
+
+```sql
+ADMIN SHOW TELEMETRY;
+```
+
+若 `DATA_PREVIEW` 列为空，说明遥测没有开启，否则说明遥测已开启。还可以从 `LAST_STATUS` 列了解上次分享使用情况信息的时间、是否成功等。
+
+对于 TiUP 遥测，可通过执行以下命令查看遥测状态：
+
+{{< copyable "shell" >}}
```
Comment (TomShawn) [inline telemetry.md:282]:
```suggestion
{{< copyable "shell-regular" >}}
```

**Pull request merged**